nexus/src/app/utilization.rs

```diff
@@ -30,4 +33,55 @@ impl super::Nexus {
     ) -> ListResultVec<db::model::SiloUtilization> {
         self.db_datastore.silo_utilization_list(opctx, pagparams).await
     }
+
+    pub async fn ip_pool_utilization_view(
+        &self,
+        opctx: &OpContext,
+        pool_lookup: &lookup::IpPool<'_>,
+    ) -> Result<IpPoolUtilization, Error> {
+        let (.., authz_pool) =
+            pool_lookup.lookup_for(authz::Action::Read).await?;
+
+        let allocated = self
+            .db_datastore
+            .ip_pool_allocated_count(opctx, &authz_pool)
+            .await?;
+        let capacity = self
+            .db_datastore
+            .ip_pool_total_capacity(opctx, &authz_pool)
+            .await?;
+
+        // we have one query for v4 and v6 allocated and one for both v4 and
+        // v6 capacity so we can do two queries instead 4, but in the response
+        // we want them paired by IP version
+        Ok(IpPoolUtilization {
+            ipv4: Ipv4Utilization {
+                // This one less trivial to justify than the u128 conversion
+                // below because an i64 could obviously be too big for u32.
+                // In addition to the fact that it is unlikely for anyone to
+                // allocate 4 billion IPs, we rely on the fact that there can
+                // only be 2^32 IPv4 addresses, period.
+                allocated: u32::try_from(allocated.ipv4).map_err(|_e| {
+                    Error::internal_error(&format!(
+                        "Failed to convert i64 {} IPv4 address count to u32",
+                        allocated.ipv4
+                    ))
+                })?,
```

Comment:
I'm slightly nervous about this but I feel like it's going to be really, really hard to hit this 500!